common/persistence/visibility/store/elasticsearch/visibility_store.go

```diff
@@ -730,11 +744,26 @@ func (s *visibilityStore) convertQuery(
 	return namespaceFilterQuery, fieldSorts, nil
 }
 
-func (s *visibilityStore) setDefaultFieldSort(fieldSorts []*elastic.FieldSort) []elastic.Sorter {
+func (s *visibilityStore) getScanFieldSorter(fieldSorts []*elastic.FieldSort) ([]elastic.Sorter, error) {
+	// if sortByDoc is set (which means ScanWorkflowExecutions is invoked), custom
+	// ORDER BY is not supported.
+	if len(fieldSorts) > 0 {
+		return nil, serviceerror.NewInvalidArgument("ORDER BY clause is not supported")
+	}
 	if len(fieldSorts) == 0 {
-		return defaultSorter
+		return docSorter, nil
 	}
+	return s.getFieldSorter(fieldSorts)
+}
+
+func (s *visibilityStore) getListFieldSorter(fieldSorts []*elastic.FieldSort) ([]elastic.Sorter, error) {
+	if len(fieldSorts) == 0 {
+		return defaultSorter, nil
+	}
+	return s.getFieldSorter(fieldSorts)
+}
 
+func (s *visibilityStore) getFieldSorter(fieldSorts []*elastic.FieldSort) ([]elastic.Sorter, error) {
```

Comment:
can we remove error from return list? this method does not return error in any case.

Reply:
Done